src/transformers/tokenization_tapas.py

```diff
@@ -323,6 +326,823 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
+            query_tokens = self.tokenize(raw_query)
+        else:
+            table_data = kwargs["table_data"]
+            query_tokens = kwargs["query_tokens"]
+
+        total_len = (
+            len(query_ids) + len(table_ids) + (self.num_special_tokens_to_add(pair=True) if add_special_tokens else 0)
+        )
+
+        overflowing_tokens = []
+        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and max_length and total_len > max_length:
+            query_ids, table_ids, overflowing_tokens = self.truncate_sequences(
+                query_ids,
+                pair_ids=table_ids,
+                num_tokens_to_remove=total_len - max_length,
+                truncation_strategy=truncation_strategy,
+                stride=stride,
+            )
+
+        if return_overflowing_tokens:
+            encoded_inputs["overflowing_tokens"] = overflowing_tokens
+            encoded_inputs["num_truncated_tokens"] = total_len - max_length
+
+        if add_special_tokens:
+            input_ids = self.build_inputs_with_special_tokens(query_ids, table_ids)
+        else:
+            input_ids = query_ids + table_ids
+
+        encoded_inputs["input_ids"] = input_ids
+
+        segment_ids = self.create_segment_token_type_ids_from_sequences(query_ids, table_data)
+        column_ids = self.create_column_token_type_ids_from_sequences(query_ids, table_data)
+        row_ids = self.create_row_token_type_ids_from_sequences(query_ids, table_data)
+        prev_label_ids = [0] * len(row_ids)
+
+        column_ranks, inv_column_ranks, columns_to_numeric_values = self._get_numeric_column_ranks(
+            column_ids, row_ids, raw_table
+        )
+        numeric_relations = self._get_numeric_relations(
+            raw_query, column_ids, row_ids, raw_table, columns_to_numeric_values
+        )
+
+        # Load from model defaults
+        if return_token_type_ids is None:
+            return_token_type_ids = "token_type_ids" in self.model_input_names
+        if return_attention_mask is None:
+            return_attention_mask = "attention_mask" in self.model_input_names
+
+        if return_attention_mask:
+            attention_mask = self.create_attention_mask_from_sequences(query_ids, table_data)
+            encoded_inputs["attention_mask"] = attention_mask
+
+        if answer_coordinates is not None and answer_text is not None:
+            label_ids = self.get_answer_ids(
+                column_ids, row_ids, table_data, query_tokens, answer_text, answer_coordinates
+            )
+            numeric_values = self._get_numeric_values(raw_table, column_ids, row_ids, columns_to_numeric_values)
+            numeric_values_scale = self._get_numeric_values_scale(raw_table, column_ids, row_ids)
+
+            encoded_inputs["label_ids"] = label_ids
+            encoded_inputs["numeric_values"] = numeric_values
+            encoded_inputs["numeric_values_scale"] = numeric_values_scale
```

Comment:
These are created, but no padding/truncation is applied to them. This currently results in an error (when I run a Colab notebook from this branch):

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/transformers/tokenization_utils_base.py in convert_to_tensors(self, tensor_type, prepend_batch_axis)
    607 
--> 608                 tensor = as_tensor(value)
    609 

ValueError: expected sequence of length 41 at dim 1 (got 45)

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
6 frames
/usr/local/lib/python3.6/dist-packages/transformers/tokenization_utils_base.py in convert_to_tensors(self, tensor_type, prepend_batch_axis)
    623                     )
    624                 raise ValueError(
--> 625                     "Unable to create tensor, you should probably activate truncation and/or padding "
    626                     "with 'padding=True' 'truncation=True' to have batched tensors with the same length."
    627                 )

ValueError: Unable to create tensor, you should probably activate truncation and/or padding with 'padding=True' 'truncation=True' to have batched tensors with the same length.
```

Reply:
Padding has been fixed